tests/test_encoder_decoder_text_summarization.py

```diff
@@ -0,0 +1,96 @@
+import json
+import re
+import subprocess
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+import pytest
+
+from .test_examples import TIME_PERF_FACTOR
+
+
+MODELS_TO_TEST = {
+    "bf16": [
+        ("facebook/bart-large-cnn", 6.643, 32.399, 2, 1),
+    ],
+    # "deepspeed": [
+    #     ("bigscience/bloomz-7b1", 27.34439410425298),
+    # ],
+}
+
+
+def _test_text_summarization(
+    model_name: str,
+    baseline: float,
+    baseline_acc: float,
+    batch_size: int,
+    num_beams: int,
+    token: str,
+    deepspeed: bool = False,
+    world_size: int = 8,
+):
+    command = ["python3"]
+    path_to_example_dir = Path(__file__).resolve().parent.parent / "examples"
+
+    if deepspeed:
+        command += [
+            f"{path_to_example_dir / 'gaudi_spawn.py'}",
+            "--use_deepspeed",
+            f"--world_size {world_size}",
+        ]
+
+    command += [
+        f"{path_to_example_dir / 'summarization' / 'run_summarization.py'}",
+        f"--model_name_or_path {model_name}",
+        "--do_predict",
+        "--predict_with_generate",
+        "--dataset_name cnn_dailymail",
+        "--dataset_config 3.0.0",
+        "--use_habana",
+        f"--per_device_eval_batch_size {batch_size}",
+        "--gaudi_config_name Habana/bart",
```

Comment:
Make this generic so we can use this for t5-3b configuration also.

Reply:
Yeah, you can add a new element in `MODELS_TO_TEST`:
```py
("facebook/bart-large-cnn", "Habana/bart", 6.643, 32.399, 2, 1),
```